lib/discovery/discovery.go

```diff
@@ -153,29 +153,18 @@ func (d *RoutingDiscovery) DiscoverAndSetFlags(
 			d.labelAnnotationPrefix,
 		)
 		for {
-			var event event
-			select {
-			case <-ctx.Done():
-				return nil
-			case r := <-watchResultCh:
-				if r.err != nil {
-					d.logger.WithError(err).Error("Error on reading event stream. Retrying stream...")
-					break
```

Comment:
This is annoying.. Braking in select just goes out out select itself, not from for loop like we expect (: